Modules/pyexpat.c

```diff
@@ -1631,101 +1627,152 @@ static int init_handler_descrs(void)
        appropriately.
     */
-    PyModule_AddStringConstant(m, "native_encoding", "UTF-8");
+    if (PyModule_AddStringConstant(mod, "native_encoding", "UTF-8") < 0) {
+        goto error;
+    }
 
-    d = PyModule_GetDict(m);
+    PyObject *d = PyModule_GetDict(mod);
     if (d == NULL) {
-        Py_DECREF(m);
-        return NULL;
+        goto error;
+    }
+
+    PyObject *errmod_name = PyUnicode_FromString(MODULE_NAME ".errors");
+    if (errmod_name == NULL) {
+        goto error;
     }
-    errors_module = PyDict_GetItemWithError(d, errmod_name);
+
+    PyObject *errors_module = PyDict_GetItemWithError(d, errmod_name);
     if (errors_module == NULL && !PyErr_Occurred()) {
         errors_module = PyModule_New(MODULE_NAME ".errors");
         if (errors_module != NULL) {
-            _PyImport_SetModule(errmod_name, errors_module);
+            if (_PyImport_SetModule(errmod_name, errors_module) < 0) {
+                Py_DECREF(errors_module);
+                Py_CLEAR(errmod_name);
+                goto error;
+            }
             /* gives away the reference to errors_module */
-            PyModule_AddObject(m, "errors", errors_module);
+            if (PyModule_AddObject(mod, "errors", errors_module) < 0) {
+                Py_DECREF(errors_module);
+                Py_CLEAR(errmod_name);
+                goto error;
+            }
         }
     }
-    Py_DECREF(errmod_name);
-    model_module = PyDict_GetItemWithError(d, modelmod_name);
+    Py_CLEAR(errmod_name);
+
+    PyObject *modelmod_name = PyUnicode_FromString(MODULE_NAME ".model");
+    if (modelmod_name == NULL) {
+        goto error;
+    }
+
+    PyObject *model_module = PyDict_GetItemWithError(d, modelmod_name);
     if (model_module == NULL && !PyErr_Occurred()) {
         model_module = PyModule_New(MODULE_NAME ".model");
         if (model_module != NULL) {
-            _PyImport_SetModule(modelmod_name, model_module);
+            if (_PyImport_SetModule(modelmod_name, model_module) < 0) {
+                Py_DECREF(model_module);
+                Py_CLEAR(modelmod_name);
+                goto error;
+            }
             /* gives away the reference to model_module */
-            PyModule_AddObject(m, "model", model_module);
+            if (PyModule_AddObject(mod, "model", model_module) < 0) {
+                Py_DECREF(model_module);
+                Py_CLEAR(modelmod_name);
+                goto error;
+            }
         }
     }
-    Py_DECREF(modelmod_name);
+    Py_CLEAR(modelmod_name);
+
     if (errors_module == NULL || model_module == NULL) {
         /* Don't core dump later! */
-        Py_DECREF(m);
-        return NULL;
+        goto error;
     }
 
 #if XML_COMBINED_VERSION > 19505
     {
         const XML_Feature *features = XML_GetFeatureList();
         PyObject *list = PyList_New(0);
-        if (list == NULL)
+        if (list == NULL) {
             /* just ignore it */
             PyErr_Clear();
```

Comment:
Again, I prefer to report the error (goto error) rather than ignoring it silently.